### ISSUE TITLE
A certain multibyte utf-character confuses function list

### ISSUE THREAD

**Issue by tm8544** [notepad-plus-plus/notepad-plus-plus#14850]
Having this line of code in one of my php files confuses functionlist, it does not list all functions if the line is active.
define("s_IKONI_MUISTIO", "📝");

This does not happen in all files with that line, I have seen this with only one file that is attached (Functionlist_test.zip).
See line #168.

Images of functionlist output are also attached with line 168 commented and not commented.
[Functionlist_test.zip](https://github.com/notepad-plus-plus/notepad-plus-plus/files/14549527/Functionlist_test.zip)
![Line 168 not commented out and only some functions visible](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/15167679/9d89c897-216c-462a-ab5e-495a845a0cfe)
![Line 168 commented out and all functions visible](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/15167679/508719d9-1135-4061-9182-593ca0e838cf)

**Comment by gab12** [comment]
Yes I have the same problem. As soon as we add an emojicon to my php code, the functionList is automatically lost. It no longer recognizes 70% of the functions but strangely 30% is displayed correctly.

Did you find a solution?

**Comment by tm8544** [comment]
Yes I did.  I changed my code as follows:

Old code causing problem:
```
define("s_IKONI_OK", "✅");
define("s_IKONI_HUOMAUTUS", "⚠️");
define("s_IKONI_VAROITUS", "⛔");
define("s_IKONI_MUISTIO", "📝");
```

New code without problem:
```
define("s_IKONI_OK", "\u{2705}");
define("s_IKONI_HUOMAUTUS", "\u{26A0}\u{FE0F}");
define("s_IKONI_VAROITUS", "\u{26D4}\u{FE0F}");
define("s_IKONI_MUISTIO", "\u{1F4DD}");
```

Emojicons stay the same, they are just expressed in a different way.

I used these when I searched ccodes for those emojicons
- https://www.compart.com/en/unicode/
- https://decodeunicode.org/en/
- https://www.iemoji.com/

**Comment by gab12** [comment]
you are the best. 
Thank you so much. This will help me a lot because it was annoying not to have the functions. 🙏🏼🙏🏼 (oups an Emojicons 😉)
